app/javascript/mastodon/components/short_number.tsx

```diff
@@ -0,0 +1,111 @@
+import { memo } from 'react';
+
+import { FormattedMessage, FormattedNumber } from 'react-intl';
+
+import { toShortNumber, pluralReady, DECIMAL_UNITS } from '../utils/numbers';
+
+/**
+ * Custom renderer for numbers
+ * @param {JSX.Element} displayNumber Number to display
+ * @param {number} pluralReady Number used for pluralization
+ * @returns {JSX.Element} Final render of number
+ */
+interface ShortNumberRenderer {
+  (displayNumber: JSX.Element, pluralReady: number): JSX.Element;
+}
+
+/**
+ * Component that renders short big number to a shorter version
+ * @param {ShortNumberProps} props  Props for the component
+ * @param {number} props.value value Number to display in short variant
+ * @param {ShortNumberRenderer} [props.renderer]
+ * Custom renderer for numbers, provided as a prop. If another renderer
+ * passed as a child of this component, this prop won't be used.
+ * @param {ShortNumberRenderer} [props.children]
+ * Custom renderer for numbers, provided as a child. If another renderer
+ * passed as a prop of this component, this one will be used instead.
+ * @returns {JSX.Element} - Rendered number.
+ */
+interface ShortNumberProps {
+  value: number;
+  renderer?: ShortNumberRenderer;
+  children?: ShortNumberRenderer;
+}
+export const ShortNumberRenderer: React.FC<ShortNumberProps> = ({
+  value,
+  renderer,
+  children,
+}): JSX.Element => {
```

Comment:
You do not need the `JSX.Element` here, `React.FC` already types it correctly

Reply:
Exactly. Deleted.